app/src/main/java/com/mapbox/maps/testapp/examples/LargeGeojsonPerformanceActivity.kt

```diff
@@ -27,21 +25,29 @@ import com.mapbox.maps.testapp.examples.annotation.AnnotationUtils
  */
 class LargeGeojsonPerformanceActivity : AppCompatActivity() {
 
-  private lateinit var routePoints: Feature
+  private val routePoints by lazy {
+    Feature.fromJson(
+      AnnotationUtils.loadStringFromAssets(
+        this@LargeGeojsonPerformanceActivity, LARGE_GEOJSON_ASSET_NAME
+      )!!
+    )
+  }
+  private var jsonUpdateCounter = 0
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     val mapView = MapView(this)
     setContentView(mapView)
 
-    routePoints = Feature.fromJson(
-      AnnotationUtils.loadStringFromAssets(
-        this@LargeGeojsonPerformanceActivity, LARGE_GEOJSON_ASSET_NAME
-      )!!
-    )
-
     mapView.getMapboxMap()
       .apply {
+        addOnSourceDataLoadedListener {
+          if (it.dataId.isNullOrBlank().not()) {
+            // log whenever update with corresponding data-id is rendered on the map
+            logI(TAG, "GeoJsonSource was updated, data-id : ${it.dataId}")
```

Comment:
Is this something we always want the user to see?

Reply:
It's for the demo app to show the api usage, should be fine I think